front/src/applications/editor/tools/routeEdition/components/RouteEditionPanel.tsx

```diff
@@ -0,0 +1,314 @@
+      });
+      if (!infraID) throw new Error('No infra selected');
+      const payload = await getCompatibleRoutesPayload(
+        infraID,
+        route.properties.entry_point,
+        route.properties.exit_point,
+        dispatch
+      );
+      const candidates = await postPathfinding({
+        infraId: infraID,
+        pathfindingInput: payload,
+      }).unwrap();
+
+      const candidateColors = chroma
+        .scale(['#321BF7CC', '#37B5F0CC', '#F0901FCC', '#F7311BCC', '#D124E0CC'])
+        .mode('lch')
+        .colors(candidates.length)
+        .map((str) => chroma(str).hex());
+
+      const features = await getRouteGeometries(
+        infraID as number,
+        route.properties.entry_point,
+        route.properties.exit_point,
+        candidates,
+        dispatch
+      );
+
+      setState({
+        optionsState: {
+          type: 'options',
+          options: candidates.map((candidate, i) => ({
+            data: candidate,
+            color: candidateColors[i],
+            feature: { ...features[i], properties: { color: candidateColors[i], index: i } },
+          })),
+        },
+      });
+    } catch (e) {
+      setState({ optionsState: { type: 'idle' } });
+    }
+  }, [infraID, setState, state.entity]);
+
+  /**
+   * Funciton that select a new route search item
+   */
+  const selectSearchRouteItem = useCallback(
+    (index: number) => {
+      setState((prev) => {
+        if (prev.optionsState.type !== 'options' || !prev.entity) return prev;
+
+        const isSelection = index !== prev.optionsState.focusedOptionIndex;
+        const candidate = prev.optionsState.options[index];
+        const route: RouteEntity = {
+          ...prev.entity,
+          properties: {
+            ...prev.entity.properties,
+            ...(isSelection
+              ? {
+                  entry_point_direction: candidate.data.track_ranges[0].direction,
+                  switches_directions: candidate.data.switches_directions,
+                  release_detectors: candidate.data.detectors,
+                }
+              : {}),
+          },
+        };
+        return {
+          ...prev,
+          entity: route,
+          isComplete: isSelection,
+          optionsState: {
+            ...prev.optionsState,
+            focusedOptionIndex: isSelection ? index : undefined,
+          },
+        };
+      });
+    },
+    [setState]
+  );
+
+  /**
+   * Function to change a route extremity.
+   */
+  const extremetyUpdate = useCallback(
+    (waypoint: WayPointEntity | null, endPoint: EndPoint) => {
+      setState((prev) => ({
+        ...prev,
+        entity: {
+          ...prev.entity,
+          properties: {
+            ...prev.entity.properties,
+            [EndPointKeys[endPoint]]: waypoint
+              ? {
+                  type: waypoint.objType,
+                  id: waypoint.properties.id,
+                }
+              : { type: 'Dector', id: NEW_ENTITY_ID },
```

Comment:
I think ? There is no type safety ?

```suggestion
              : { type: 'Detector', id: NEW_ENTITY_ID },
```